runtime/include/map.h

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "lock.h"
+
+/* Simple K-V store based on The Practice of Programming by Kernighan and Pike */
+
+/* Bucket count is sized to be a prime that is approximately 20% larger than the desired capacity (6k keys) */
+#define MAP_BUCKET_COUNT 7907
+#define MAP_LOCK_STRIPES 17
+
+#define MAP_HASH jenkins_hash
+
+struct map_node {
+	uint32_t         hash;
+	uint8_t         *key;
+	uint32_t         key_len;
+	uint8_t         *value;
+	uint32_t         value_len;
+	struct map_node *next;
+};
```

Comment:
The ordering of these will waste 96 bytes given alignment rules. Not sure if it is that much less readable if you pt the 32 bit values contiguously.